modules/renter/hostdb/scan.go

```diff
@@ -136,10 +136,16 @@ func (hdb *HostDB) threadedProbeHosts() {
 	for hostEntry := range hdb.scanPool {
 		// Request settings from the queued host entry.
 		// TODO: use dialer.Cancel to shutdown quickly
-		hdb.log.Debugln("Scanning", hostEntry.NetAddress, hostEntry.PublicKey)
+		//
+		// A readlock is necessary when viewing the elements of the host entry.
+		hdb.mu.RLock()
+		netAddr := hostEntry.NetAddress
+		pubKey := hostEntry.PublicKey
+		hdb.mu.RUnlock()
```

Comment:
The netaddress is allowed to change, a host can change IP addresses or domain names, as the renter is supposed to recognize the host by its pubkey.

Currently the renter does not, but this is a preparation for the day that it does.